src/abstractdataframe/reshape.jl

```diff
@@ -244,6 +244,7 @@ end
 
 function _unstack(df::AbstractDataFrame, rowkeys::AbstractVector{Symbol},
                   colkey::Int, value::Int, keycol, valuecol, g, renamecols)
+    g.idx === nothing && compute_indices!(g)
```

Comment:
maybe it would be cleaner to define `indices` (or some more appropriate name) unexported function that we would use instead of `g.idx` everywhere and it would do this check internally. This should be a safer design in the long therm I think.

Reply:
You mean move the check inside `compute_indices!`?

Reply:
even more. I would define the following for example:
```
function Base.getproperty(g::GroupedDataFrame, f::Symbol)
    f === :idx && getfield(g, :idx) === nothing && compute_indices!(g)
    getfield(g, f)
end
```
in this way future developers will not have to think about checking if `:idx` was computed or not.

Reply:
of course we do not have to do it.

Reply:
Yeah I had quickly considered this option. It's appealing. If we do that, it should also apply to `starts` and `ends` fields.

Reply:
Well - please decide what you prefer. It is an internal issue, and you are the lead developer of this part of code 😄.